### ISSUE TITLE
Failed to load plugin dplay. ImportError: cannot import name 'quote'

### ISSUE THREAD

**Issue by thinkpad4** [streamlink/streamlink#3291]
I FIXED IT, IT WAS MY FAULT FOR NOT TROUBLESHOOTING MORE.  Sorry

## Bug Report

- [x] This is a bug report and I have read the contribution guidelines.

### Description

When I run Streamlink from command prompt I get a error at the start before the stream loads

### Expected / Actual behavior

I expect it not to show that error at the start, BUT it does still load the stream and plays it normally

### Reproduction steps / Explicit stream URLs to test

Even just running Streamlink allow causes the error to appear
1. ...
2. ...
3. ...

### Log output

```
Failed to load plugin dplay:
  File "imp.py", line 235, in load_module
  File "imp.py", line 172, in load_source
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "d:\Program Files (x86)\Streamlink\pkgs\streamlink\plugins\dplay.py", lin
e 6, in <module>
    from streamlink.compat import quote
ImportError: cannot import name 'quote'

[cli][info] Found matching plugin twitch for URL https://www.twitch.tv/gabbykayy best
```

### Additional comments, etc.

I am using this Dev version: streamlink-1.7.0_28.g777b60f-win32

[Love Streamlink? Please consider supporting our collective. Thanks!](https://opencollective.com/streamlink/donate)

**Comment by bastimeyer** [comment]
I've unlocked and re-opened this thread, because of what was reported in streamlink/streamlink-twitch-gui#781.

It looks like the `dplay` plugin wasn't included in the removed plugins list when it was first implemented in #2003. This and any other plugin which were not properly included in that list need to be added to `src/streamlink/plugins/.removed` in order to fix this issue when upgrading from a very old Streamlink install on Windows. The better approach would be fixing the installer though, so that it always uninstalls Streamlink first.

**Comment by bastimeyer** [comment]
Hm...

```bash
diff --side-by-side -W 60 \
  <(tail -n+4 src/streamlink/plugins/.removed | sort -u) \
  <(for file in $(git log --all --pretty=format: --name-only --diff-filter=D src/streamlink/plugins/ | sort -u); do basename "$file" .py; done)
```
```diff
afreecatv                       afreecatv
aftonbladet                     aftonbladet
aliez                           aliez
alieztv                      <
aljazeeraen                     aljazeeraen
antenna                         antenna
apac                            apac
arconai                         arconai
azubutv                         azubutv
bambuser                        bambuser
beam                         <
beattv                          beattv
bliptv                          bliptv
bongacams                       bongacams
brittv                          brittv
cam4                            cam4
camsoda                         camsoda
canlitv                         canlitv
chaturbate                      chaturbate
connectcast                     connectcast
cubetv                          cubetv
cybergame                       cybergame
cyro                            cyro
daisuki                         daisuki
dingittv                        dingittv
disney_de                       disney_de
dmcloud                      |  dlive
dmcloud_embed                   dmcloud_embed
douyutv                      |  dmcloud
douyutv_blackbox                douyutv_blackbox
                             >  douyutv
                             >  dplay
ellobo                          ellobo
eurocom                         eurocom
europaplus                      europaplus
expressen                       expressen
filmon_us                       filmon_us
                             >  foxnews
furstream                       furstream
gaminglive                      gaminglive
gomexp                          gomexp
                             >  huomao
                             >  idmantv
itvplayer                       itvplayer
kanal7                          kanal7
kingkong                        kingkong
                             >  kralmuzik
letontv                         letontv
livecodingtv                    livecodingtv
livestation                     livestation
looch                           looch
media_ccc_de                    media_ccc_de
meerkat                         meerkat
metube                          metube
mico                            mico
mips                            mips
mixer                           mixer
mlgtv                           mlgtv
neulion                         neulion
nineanime                       nineanime
npo                             npo
                             >  ntv
ok_live                         ok_live
oldlivestream                   oldlivestream
ovvatv                          ovvatv
pandatv                         pandatv
pcyourfreetv                    pcyourfreetv
reshet                          reshet
rte                             rte
seemeplay                       seemeplay
seetv                           seetv
servustv                        servustv
skai                            skai
speedrunslive                   speedrunslive
srgssr                          srgssr
startv                          startv
stream                       <
streamboat                      streamboat
streamlive                      streamlive
streamme                        streamme
                             >  stream
streamupcom                     streamupcom
tamago                          tamago
                             >  toya
trt                             trt
trtspor                         trtspor
tv1channel                      tv1channel
tv8cat                          tv8cat
tvcatchup                       tvcatchup
tvnbg                           tvnbg
ufctv                        <
vaughnlive                      vaughnlive
veetle                          veetle
vgtv                            vgtv
viagame                         viagame
viasat_embed                    viasat_embed
wattv                           wattv
weeb                            weeb
willax                          willax
younow                          younow
```
